### DISCUSSION TITLE
Unable to run `celery migrate` between redis source and rabbitmq (with prioritization enabled) destination. Error `Queue.declare: (406) PRECONDITION_FAILED - inequivalent arg 'x-max-priority' for queue`

### DISCUSSION THREAD

**Discussion by kcharwood** [celery/celery#9123] [Q&A] [answered]
Hello!

I am attempting to migrate from redis to rabbitmq in order to get full support for prioritization for tasks.

In looking at the [`celery migrate`](https://docs.celeryq.dev/en/stable/reference/cli.html#celery-migrate), it seems to me like I should be able to do an in-place upgrade, and once my workers are configured to consume from the rabbitmq broker, run the migration against the redis broker, and move over all the uncompleted tasks.

New tasks work fine, so I can confirm my infrastructure is good.

However, I'm running into an issue when I run `celery migrate`. I'm getting the following error:

> amqp.exceptions.PreconditionFailed: Queue.declare: (406) PRECONDITION_FAILED - inequivalent arg 'x-max-priority' for queue 'worker_queue' in vhost '/': received the value '10' of type 'signedint' but current is none

Can anyone shed any light into this error or how I am be approaching this incorrectly?

**Comment by kcharwood** [comment, marked as answer]
I was able to debug the issue, and develop a work around.

In my scenario, I'm moving from a redis broker with no priority to a rabbitmq broker with prioritization, and out of the box `migrate` does not seem to support this specific migration.

I was able to trace it to this section of code in the migrate path (specifically the error was being thrown from the `declare()` method.

https://github.com/celery/celery/blob/2859554a4707e53a1cdfe79dc073f5a8c4348f21/celery/contrib/migrate.py#L106-L115

In my configuration, I am moving to queues that have max-priority set to 10. By making the following changes, I was able to get the migration to complete.

```python
    def on_declare_queue(queue):
        new_queue = queue(producer.channel)

        # Hardcode these values to bypass the precondition check
        new_queue.max_priority = 10
        new_queue.queue_arguments = {"x-max-priority": 10}

        new_queue.name = queues.get(queue.name, queue.name)
        if new_queue.routing_key == queue.name:
            new_queue.routing_key = queues.get(queue.name,
                                               new_queue.routing_key)
        if new_queue.exchange.name == queue.name:
            new_queue.exchange.name = queues.get(queue.name, queue.name)
        new_queue.declare()
```

This allowed the migration to finish.

Shoutout to @kayluhb with the pair programming assist.

**Reply by Nusnus** [reply]
Thanks for sharing your solution!
Good work 👏
